base/mpfr.jl

```diff
@@ -504,6 +504,9 @@ end
 ^(x::BigFloat, y::Integer)  = typemin(Clong)  <= y <= typemax(Clong)  ? x^Clong(y)  : x^BigInt(y)
 ^(x::BigFloat, y::Unsigned) = typemin(Culong) <= y <= typemax(Culong) ? x^Culong(y) : x^BigInt(y)
 
+# override default inlining of x^2 etc.
+^{p}(x::BigFloat, ::Type{Val{p}}) = x^Culong(p)
```

Comment:
this gives an `InexactError` for negative powers, such as `BigFloat(2) ^ -20` (which was amazingly not tested)

(wrong line beforehand)